### PR TITLE
WA-257 Onboarding pages

### PR DESCRIPTION
**Pull request by germartinez**
Creation of the onboarding pages:
- **Welcome page** to show a text explaining the purpose of the chrome extension.
- **Download apps page** to show two QR-codes to download the android and iOS mobile apps.

It was configured the size of the QR-codes and added the Gnosis logo in the header of the chrome extension.

Added CSS-in-modules

### CODE REVIEW

File: app/components/App/styles.css [hunk 1]
```diff
@@ -85,6 +94,21 @@ button {
   width: 60px;
 }
 
+.download-apps {
```
Comment (apanizo) [inline app/components/App/styles.css]: General CSS classes should be avoided if possible when app grows the risk of overriding styles is higher. 

I strongly suggest to move the specific css to each folder. For example if the ```DownloadApps/index.js``` uses  ```download-app``,  you should create a `DownloadApps/index.css``` and then inside ```DownloadApps/index.js``` you should call:

```import `index.css```

Once you have redistributed the CSS, next step is achieve "CSS-in-modules", basically, with that technique, you assure not other styles are affecting the specific css.

After it, you should be able to use ```scss``` variables. Keep in mind when designers create proper mocks to you you have to update the views as fast as possible.

For doing the css-in-modules stuff, you have to install different webpack loaders and configure them. I suggest to take a look into ```gnosis-team-safe``` project. There you will find everything (even the configuration for using webpack 4).

File: app/components/DownloadApps/index.js [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+  }
+
+  handleOpenApp = (url) => (e) => {
+    chrome.tabs.create({ url })
+  }
+
+  render() {
+    return (
+      <div>
+        <Header />
```
Comment (apanizo) [inline app/components/DownloadApps/index.js]: The pattern of using:
```
<div>
    <Header />
     ....
<div>
```

Has been used once per view, you should create a "layout" component describing it. The main idea behind it is:
* DRY
* Have a component where if you want to modify the general aspect of a view you only have to do once

Just FYI, In the ```gnosis-team-safe``` I have created the ```PageFrame```, but for you just:
```
type Props = {
  chldren: React.Node,
}
const PageFrame = ({children}: Props) => (
  <div>
    <Header />
    <main>
        { children }
    </main>
  </div>
)

export default PageFrame
```
should be enough. Let me know if you have doubts or you want to implement the "semantic" definition of views.

**Comment by apanizo** [comment]
Any quesiton just ping me ;)

File: app/components/Welcome/index.js [hunk 3]
```diff
@@ -0,0 +1,24 @@
+import React from 'react'
+import { Link } from 'react-router-dom'
+
+import ContentPage from 'components/ContentPage'
+import gnosisOwl from 'assets/gnosis_owl.svg'
+import styles from './index.css'
+
+const Welcome = () => (
+  <ContentPage>
```
Comment (apanizo) [inline app/components/Welcome/index.js]: Why you do not use ```<Page>``` here?

File: app/components/Welcome/index.js [hunk 4]
```diff
@@ -0,0 +1,24 @@
+import React from 'react'
```
Comment (apanizo) [inline app/components/Welcome/index.js]: Also I recommend for all routes to place them in a ```routes``` folder. Following the next schema:
```
src/routes/welcome
    + components
           + Layout.tsx
           + MyPrivateComponent
                    +index.jsx
                    + index.scss
       + store
                + (actions | selectors | reducers | models) / If you need them
        index.jsx //React container
        selector.js //Selectors used in the container done with reselect
        actions.js //Basically the mapToDispatch object  used in the container
```

In that way you have isolated the specific code of routes, easier to maintain, easier to debug.

Comment (apanizo) [reply app/components/Welcome/index.js]: Of course, that is a "complex" example, if in your route you do not need anything from the store, you can skip the **store** folder (etc...)

File: app/components/DownloadApps/index.js [hunk 5]
```diff
@@ -0,0 +1,101 @@
+import React, { Component } from 'react'
```
Comment (apanizo) [inline app/components/DownloadApps/index.js]: The same comment I wrote to "Welcome" page applies here as well.
General folders like src/components or utils should be kept them as smaller and simpler as possible. The main idea is scalability and maintainability of the project.

File: app/components/CreateAccount/index.js [hunk 6]
```diff
@@ -4,7 +4,7 @@ import HdKey from 'ethereumjs-wallet/hdkey'
 import Bip39 from 'bip39'
```
Comment (apanizo) [inline app/components/CreateAccount/index.js]: And... same here :)

File: app/components/Welcome/index.js (hunk 3 above)
Comment (germartinez) [reply app/components/Welcome/index.js]: ContentPage do not display the header

Comment (apanizo) [reply app/components/Welcome/index.js]: in that case I recommend to do something:
```
<Page withoutHeader ...>
    ....
</Page>
```
And then, inside Page:
```
const Page = ({withoutHeader = false}) => (
  <div ...>
      { !withoutHeader && <Header ..../> }
      ....
  </div>
```
All pages routes/pages should follow the same layout. In that way, if you apply a change to Page component you know all routes will be updated. 

Keep the good job.

File: app/routes/Account/components/Layout.js [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+    const {
+      address,
+      network,
+      handleShowQrCode,
+    } = this.props
+
+    return (
+      <Page
+        logOut={true}
+        whitelist={true}
```
Comment (apanizo) [inline app/routes/Account/components/Layout.js]: You can write boolean attributes when they are true like:
```
<Page logOut whitelist>
    ....
</Page>
```

File: app/routes/Account/components/Layout.js [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+      >
+        <p>Address: {address}</p>
+        <p>Network: {network}</p>
+
```
Comment (apanizo) [inline app/routes/Account/components/Layout.js]: Why those white lines? :)

File: app/routes/ConfirmTransaction/components/Layout.js [hunk 9]
```diff
@@ -0,0 +1,42 @@
...
+          placeholder='Password'
+          value={password}
+          onChange={updatePassword}
+        />
+
+        {errorMessage && <p>{errorMessage}</p>}
+
+        <button onClick={handleConfirmTransaction}>
+          Confirm transaction
+      </button>
```
Comment (apanizo) [inline app/routes/ConfirmTransaction/components/Layout.js]: tab

**Pull request merged**